File: src/card/Suit.java
package card;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public enum Suit {
    SPADE("スペード"),
    HEART("ハート"),
    DIAMOND("ダイヤ"),
    CLUB("クラブ");

    @Getter
    private final String mark;
}

File: scala/src/main/scala/Suit.scala
sealed abstract class Suit(val mark: String)

object Suit {

  case object SPADE extends Suit("スペード")

  case object HEART extends Suit("ハート")

  case object DIAMOND extends Suit("ダイヤ")

  case object CLUB extends Suit("クラブ")

  def values = List(SPADE, HEART, DIAMOND, CLUB)
}


